tools/rebuild-image

```diff
@@ -40,13 +40,27 @@ commit_timestamp() {
 	git show -s --format=%ct "$rev"
 }
 
+# is the SHA1 actually present in the repo?
+# it could be it isn't, e.g. after a force push
+commit_valid() {
+	local rev=$1
+	git rev-parse --quiet --verify "$rev^{commit}"
+	[ $? -eq 0 ]
+}
+
 cached_revision=$(cached_image_rev)
 if [ -z "$cached_revision" ]; then
 	echo ">>> No cached image found; rebuilding"
 	rebuild
 	exit 0
 fi
 
+if ! commit_valid "$cached_revision"; then
+  echo ">>> cached SHA1 not found in repo, rebuilding"
```

Comment:
Fix indentation to one tab instead of two spaces. Also, `s/cached/Cached` and `s/,/;`.